src/biom.cpp

```diff
@@ -309,15 +426,25 @@ unsigned int biom::get_sample_data_direct(const std::string &id, uint32_t *& cur
 }
 
 double* biom::get_sample_counts() {
-    double *sample_counts = (double*)calloc(sizeof(double), n_samples);
-    for(unsigned int i = 0; i < n_obs; i++) {
-        unsigned int count = obs_counts_resident[i];
-        uint32_t *indices = obs_indices_resident[i];
-        double *data = obs_data_resident[i];
-        for(unsigned int j = 0; j < count; j++) {
-            uint32_t index = indices[j];
-            double datum = data[j];
-            sample_counts[index] += datum;
+    double *sample_counts = (double*)malloc(sizeof(double) * n_samples);
+
+    #pragma omp parallel
+    { 
+        #pragma omp for
+        for(int i = 0; i < n_samples; i++) {
+            sample_counts[i] = 0;
+        }
+
+        #pragma omp for reduction(+ : sample_counts[:n_samples])
```

Comment:
Please check this close, this is the first time I've done a reduction. While the tests pass, which improves confidence, I think a close examination of the use of parallel here is warranted.

Reply:
Are you sure this is even legal?
reduction needs a local copy of whatever needs to be reduced.

This code seems way to convoluted to be parallelized in a naive way.
I think it needs a complete refactoring, if we really want to parallelize it.
(is it really that expensive.?)

Reply:
I thought it was a gamble.... :) The tests pass, but that doesn't mean it is a safe approach.

It is plausibly more expensive than the other operations on load (other than malloc') as its a full table scan. To be fair, I have not profiled it. If you recommend backing this out, and we defer to later, that's totally fine

Reply:
Removed